### PR TITLE
feat: schema changes for aor24

### LINKED ISSUE #56: Schema refactor

### PR DESCRIPTION
**Pull request by novice0192**

### CODE REVIEW

File: migrations/2023-12-26-155230_new_feat_2024/up.sql [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ADD  artifacts INTEGER NOT NULL DEFAULT 0;
+
+
+ALTER TABLE public.game
+DROP  robots_destroyed,
+ADD  artifacts_collected INTEGER NOT NULL;
+
+
+ALTER TABLE public.attacker_type
+ADD level_ INTEGER NOT NULL,
```
Comment (sarvesh-03) [inline migrations/2023-12-26-155230_new_feat_2024/up.sql]: why did you suffix with "_" in level column?

File: migrations/2023-12-26-155230_new_feat_2024/up.sql [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ADD cost INTEGER NOT NULL;
+
+ALTER TABLE public.defender_type
+ADD level_ INTEGER NOT NULL,
```
Comment (sarvesh-03) [inline migrations/2023-12-26-155230_new_feat_2024/up.sql]: here too..

File: migrations/2023-12-26-155230_new_feat_2024/up.sql [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ADD cost INTEGER NOT NULL;
+
+ALTER TABLE public.mine_type
+ADD level_ INTEGER NOT NULL,
```
Comment (sarvesh-03) [inline migrations/2023-12-26-155230_new_feat_2024/up.sql]: here too..

File: migrations/2023-12-26-155230_new_feat_2024/up.sql [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+ADD cost INTEGER NOT NULL;
+
+
+ALTER TABLE public.block_type RENAME TO building_type_temp;
+ALTER TABLE public.building_type_temp
+DROP  entrance_x,
+DROP  entrance_y,
+ADD level_ INTEGER NOT NULL,
```
Comment (sarvesh-03) [inline migrations/2023-12-26-155230_new_feat_2024/up.sql]: here too..

File: migrations/2023-12-26-155230_new_feat_2024/up.sql [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+DROP diffuser_type,
+DROP building_category,
+DROP blk_type,
+ADD category block_category NOT NULL,
+ADD building_type INTEGER NOT NULL,
+ADD CONSTRAINT building_type_fk FOREIGN KEY (building_type) REFERENCES public.building_type(id);
+DROP TYPE building_category;
+
+ALTER TABLE public.map_spaces
+DROP  rotation,
```
Comment (sarvesh-03) [inline migrations/2023-12-26-155230_new_feat_2024/up.sql:59]: Its better to have rotation...

File: src/api/defense/shortest_path.rs [hunk 6]
```diff
@@ -148,22 +126,11 @@ pub fn run_shortest_paths(
     let mut shortest_paths = vec![];
     for i in &mapspaces_list {
         for j in &mapspaces_list {
```
Comment (sarvesh-03) [inline src/api/defense/shortest_path.rs:127]: BE CAREFUL WITH THIS LOOP

**Review by sarvesh-03 — APPROVED** [review]
- ADD AUTH Middleware or make every route protected.
- Update rust image in Docker File

**Comment by novice0192** [comment]
> * ADD AUTH Middleware or make every route protected.
> 
>     * Update rust image in Docker File

will add auth middleware after merging oauth pr

**Pull request merged**